freud/box.pyx

```diff
@@ -301,9 +302,13 @@ cdef class Box:
         vecs = np.atleast_2d(vecs)
         vecs = freud.util._convert_array(vecs, shape=(None, 3)).copy()
 
-        cdef const float[:, ::1] l_points = vecs
-        cdef unsigned int Np = l_points.shape[0]
-        self.thisptr.wrap(<vec3[float]*> &l_points[0, 0], Np)
+        if inplace is False:
+            cdef const float[:, ::1] l_points = vecs
+            cdef unsigned int Np = l_points.shape[0]
+            self.thisptr.wrap(<vec3[float]*> &l_points[0, 0], Np)
+        else:
+            cdef unsigned int Np = vecs.shape[0]
+            self.thisptr.wrap(<vec3[float]*> &vecs[0, 0], Np)
```

Comment:
Write the "true" part first. Also don't check `is False`, just let the value be interpreted as truthy/falsy.
```suggestion
        if inplace:
            cdef unsigned int Np = vecs.shape[0]
            self.thisptr.wrap(<vec3[float]*> &vecs[0, 0], Np)
        else:
            cdef const float[:, ::1] l_points = vecs
            cdef unsigned int Np = l_points.shape[0]
            self.thisptr.wrap(<vec3[float]*> &l_points[0, 0], Np)
```